### PR TITLE
locate:js: refactor to improve readability of code.

### PR DESCRIPTION
**Pull request by jfparadis**
- remove ternary operators with "do nothing" part
- reduce the number of variables

### CODE REVIEW

**Comment by michaelfig** [comment]
Please rebase against `master`, which now properly depends on the correct version of transform-module.

Then the tests should pass.

**Comment by jfparadis** [comment]
@michaelfig Still the same TF

**Comment by michaelfig** [comment]
Now the tests are succeeding on master.  Please rebase and try again.

Thanks!

File: src/locate.js [hunk 1]
```diff
@@ -3,16 +3,19 @@ export const makeSuffixLocator = (suffix = '.js') => {
...
-    const path = url.pathname.endsWith('/')
-      ? `${url.pathname}index`
-      : url.pathname;
+    if (path.endsWith('/')) {
+      path = `${path}index`;
+    }
 
     // Translate missing suffix to have one.
-    const sfxPath = path.endsWith(suffix) ? path : `${path}${suffix}`;
+    if (!path.endsWith(suffix)) {
```
Comment (dckc) [inline src/locate.js:14]: Mutable state is easier to read than `?` expressions? Hm. not for me.

Comment (erights) [reply src/locate.js:14]: FWIW, I find the new one easier to read and understand. The "?:" form, to express that one branch is good "as is", seems no easier to understand than the extremely local update of the `path` variable. When there's no possibility of aliasing, the uses and updates are all textually very local and co-visible, and none of the variables are captured or accessed by less direct control flow, then their's no loss of reliable static reasoning compared to the other form. This was also my reasoning for proposing that Jessie objects could be mutable and mutated as long as no aliasing was yet possible.

Comment (dckc) [reply src/locate.js:14]: fair enough.

ML habits die hard, I guess. (haskell, ocaml, ...)

**Pull request merged**